internal/server/iigointernal/orchestration.go

```diff
@@ -68,87 +68,93 @@ var presidentPointer roles.President = nil
-	errWithdrawSpeaker := featurePresident.withdrawSpeakerSalary(g)
+	presidentWithdrawSuccess := judicialBranch.withdrawPresidentSalary(g)
+	judgeWithdrawSuccess := legislativeBranch.withdrawJudgeSalary(g)
+	speakerWithdrawSuccess := executiveBranch.withdrawSpeakerSalary(g)
 
 	// Handle the lack of resources
-	if errWithdrawPresident != nil {
+	if !presidentWithdrawSuccess {
+		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
 		return errors.Errorf("Could not run IIGO since President has no resoruces to spend")
 	}
-	if errWithdrawJudge != nil {
+	if !judgeWithdrawSuccess {
+		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
 		return errors.Errorf("Could not run IIGO since Judge has no resoruces to spend")
 	}
-	if errWithdrawSpeaker != nil {
+	if !speakerWithdrawSuccess {
+		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
 		return errors.Errorf("Could not run IIGO since Speaker has no resoruces to spend")
 	}
 
 	// Pay salaries into budgets
-	featureJudge.sendPresidentSalary()
-	featureSpeaker.sendJudgeSalary()
-	featurePresident.sendSpeakerSalary()
+	judicialBranch.sendPresidentSalary(&executiveBranch)
+	legislativeBranch.sendJudgeSalary(&judicialBranch)
+	executiveBranch.sendSpeakerSalary(&legislativeBranch)
 
 	// 1 Judge actions - inspect history
-	_, judgeInspectingHistoryError := featureJudge.InspectHistory()
+	_, historyInspected := judicialBranch.inspectHistory()
 
 	// 2 President actions
 	resourceReports := map[shared.ClientID]shared.Resources{}
 	var aliveClientIds []shared.ClientID
-	for _, v := range rules.VariableMap["islands_alive"].Values {
+	for _, v := range rules.VariableMap[rules.IslandsAlive].Values {
 		aliveClientIds = append(aliveClientIds, shared.ClientID(int(v)))
 		resourceReports[shared.ClientID(int(v))] = iigoClients[shared.ClientID(int(v))].ResourceReport()
 	}
-	featurePresident.broadcastTaxation(resourceReports)
-	featurePresident.requestAllocationRequest()
-	featurePresident.replyAllocationRequest(g.CommonPool)
-	featurePresident.requestRuleProposal()
-	ruleToVote := featurePresident.getRuleForSpeaker()
+	executiveBranch.broadcastTaxation(resourceReports)
+	executiveBranch.requestAllocationRequest()
+	executiveBranch.replyAllocationRequest(g.CommonPool)
+	executiveBranch.requestRuleProposal()
+	ruleToVote, ruleSelected := executiveBranch.getRuleForSpeaker()
 
 	// 3 Speaker actions
-	featureSpeaker.setRuleToVote(ruleToVote)
-	featureSpeaker.setVotingResult(iigoClients)
-	_ = featureSpeaker.announceVotingResult()
+	if ruleSelected {
+		legislativeBranch.setRuleToVote(ruleToVote)
+		legislativeBranch.setVotingResult(aliveClientIds)
+		legislativeBranch.announceVotingResult()
+	}
 
 	// 4 Declare performance (Judge) (in future all the iigointernal)
-	if judgeInspectingHistoryError != nil {
-		featureJudge.declarePresidentPerformanceWrapped()
+	if historyInspected {
```

Comment:
potentially a judge may wanna declare the performance of a speaker/president without even looking at the history, but a minor issue for not. it can be revisited

Reply:
No idea really, I think this can be pushed to post mvp